viz/visualize.py

```diff
@@ -188,19 +192,36 @@ def all_latent_traversals(self, size=8, filename='imgs/all_traversals.png'):
             Number of samples for each latent traversal.
         """
         latent_samples = []
-
+        avg_kl_list = read_avg_kl_from_file(self.model_dir + '/kl_data.log')
         # Perform line traversal of every latent
         for idx in range(self.model.latent_dim):
             latent_samples.append(self.latent_traverser.traverse_line(idx=idx,
                                                                       size=size))
+        latent_samples = [latent_sample for _, latent_sample in sorted(zip(avg_kl_list, latent_samples), reverse=True)]
+        sorted_avg_kl_list = [round(float(avg_kl_sample), 3) for avg_kl_sample, _ in sorted(zip(avg_kl_list, latent_samples), reverse=True)]
 
         # Decode samples
         generated = self._decode_latents(torch.cat(latent_samples, dim=0))
 
+        # Convert tensor to PIL Image
+        tensor = make_grid(generated.data, nrow=size, pad_value=10)
+        all_traversal_im = transforms.ToPILImage()(tensor)
+        # Resize image
+        new_width = int(1.3 * all_traversal_im.width)
+        new_size = (all_traversal_im.height, new_width)
+        traversal_images_with_text = Image.new("RGB", new_size, color='white')
+        traversal_images_with_text.paste(all_traversal_im, (0, 0))
+        # Add KL text alongside each row
+        draw = ImageDraw.Draw(traversal_images_with_text)
+        for latent_idx, latent_dim in enumerate(sorted_avg_kl_list):
+            draw.text(xy=(int(0.825 * traversal_images_with_text.width), int((latent_idx / len(sorted_avg_kl_list) + 1 / (2 * len(sorted_avg_kl_list))) * all_traversal_im.height)),
```

Comment:
all of these are extremely long lines: add enter between commas to make it approximatively pip8.

I didn't really try undertanding what you are doing tbh: but if it works then good :)